cmd/lock-rpc-server.go

```diff
@@ -250,49 +276,46 @@ func getLongLivedLocks(m map[string][]lockRequesterInfo, interval time.Duration)
 // lockMaintenance loops over locks that have been active for some time and checks back
 // with the original server whether it is still alive or not
 func (l *lockServer) lockMaintenance(interval time.Duration) {
-
 	l.mutex.Lock()
 	// get list of locks to check
 	nlripLongLived := getLongLivedLocks(l.lockMap, interval)
 	l.mutex.Unlock()
 
 	for _, nlrip := range nlripLongLived {
-
+		// Initialize client based on the long live locks.
 		c := newClient(nlrip.lri.node, nlrip.lri.rpcPath)
 
 		var active bool
 
 		// Call back to original server verify whether the lock is still active (based on name & uid)
-		if err := c.Call("Dsync.Active", &LockArgs{Name: nlrip.name, UID: nlrip.lri.uid}, &active); err != nil {
-			// We failed to connect back to the server that originated the lock, this can either be due to
+		if err := c.Call("Dsync.Active", &LockArgs{
+			Name: nlrip.name,
+			UID:  nlrip.lri.uid,
+		}, &active); err != nil {
+			// We failed to connect back to the server that originated the lock,
+			// this can either be due to
 			// - server at client down
 			// - some network error (and server is up normally)
 			//
 			// We will ignore the error, and we will retry later to get resolve on this lock
-			c.Close()
-		} else {
-			c.Close()
-
-			if !active { // The lock is no longer active at server that originated the lock
-				// so remove the lock from the map
-				l.mutex.Lock()
-				// Check if entry is still in map (could have been removed altogether by 'concurrent' (R)Unlock of last entry)
-				if lri, ok := l.lockMap[nlrip.name]; ok {
-					if !l.removeEntry(nlrip.name, nlrip.lri.uid, &lri) {
-						// Remove failed, in case it is a:
-						if nlrip.lri.writer {
-							// Writer: this should never happen as the whole (mapped) entry should have been deleted
-							log.Errorln("Lock maintenance failed to remove entry for write lock (should never happen)", nlrip.name, nlrip.lri, lri)
-						} else {
-							// Reader: this can happen if multiple read locks were active and the one we are looking for
-							// has been released concurrently (so it is fine)
-						}
-					} else {
-						// remove went okay, all is fine
-					}
-				}
-				l.mutex.Unlock()
+		}
+		c.Close()
+		if !active {
```

Comment:
With `active` initialized to false (line 288) and in case of any errors for the Dsync.Active call, we would now remove the lock (!)

Locks should only be removed when `err == nil && !active`